ceph: vault kms configuration for ceph object store (#6641)
rook/rook: Documentation/ceph-cluster-crd.md

However, if a Key Management System exists Rook is capable of using it. HashiCorp Vault is the only KMS currently supported by Rook.
Please refer to the next section.

> RGW supports encryption via KMS using HashiCorp Vault. If below settings are defined, then RGW establish connection between Vault and

Reviewer: ```suggestion
Ceph RGW supports encryption via KMS using HashiCorp Vault. If the below settings are defined, then RGW establish a connection between Vault and
```

> whenever s3 client sends request with SSE, it encrypts that using key specified by the client. For more details w.r.t RGW please

Reviewer: ```suggestion
whenever S3 client sends a request with Server Side Encryption, it encrypts that using the key specified by the client. For more details w.r.t RGW, please
```

However, if a Key Management System exists Rook is capable of using it. HashiCorp Vault is the only KMS currently supported by Rook.
Please refer to the next section.

> RGW supports encryption via KMS using HashiCorp Vault. If below settings are defined, then RGW establish connection between Vault and

Reviewer: It would be nice to link the official RGW doc for more details?

Author: I have the link at the end of the doc, it makes sense to add here

Note: if you are using self-signed certificates (not known/approved by a proper CA) you must pass `VAULT_SKIP_VERIFY: true`.
Communications will remain encrypted but the validity of the certificate will not be verified.

For RGW, please note the following:

> * `VAULT_BACKEND` is option specifically for RGW to mention about secret engine which can be used, currently supports two : kv and transit.

Reviewer: ```suggestion
* `VAULT_BACKEND` option is specifically for RGW to mention about the secret engine which can be used, currently supports two: `kv` and `transit`.

Also please link the Vault docs for those two.
```

> * The Storage administrator need to create secret in Vault server, so that s3 clients uses that key for encryption

Reviewer: ```suggestion
* The Storage administrator needs to create a secret in the Vault server so that s3 clients use that key for encryption
```

Note: if you are using self-signed certificates (not known/approved by a proper CA) you must pass `VAULT_SKIP_VERIFY: true`.
Communications will remain encrypted but the validity of the certificate will not be verified.

For RGW, please note the following:

* `VAULT_BACKEND` option is specifically for RGW to mention about the secret engine which can be used, currently supports two: [kv](https://www.vaultproject.io/docs/secrets/kv) and [transit](https://www.vaultproject.io/docs/secrets/transit).

> * The Storage administrator needs to create secret in the Vault server so that s3 clients use that key for encryption

Reviewer: ```suggestion
* The Storage administrator needs to create a secret in the Vault server so that S3 clients use that key for encryption
```

> ```hcl

Reviewer: ```suggestion
```console
```
